test/libgit2.jl

```diff
@@ -1662,6 +1740,121 @@ mktempdir() do dir
+
+                # No credential settings in configuration.
+                cred = LibGit2.GitCredential("https", "github.com")
+                username = LibGit2.filter_username(cfg, cred)
+                @test isnull(username)
+
+                # Add a credential setting for a specific for a URL
+                LibGit2.set!(cfg, "credential.https://github.com.username", "foo")
+
+                cred = LibGit2.GitCredential("https", "github.com")
+                username = LibGit2.filter_username(cfg, cred)
+                @test !isnull(username)
+                @test get(username) == "foo"
+
+                cred = LibGit2.GitCredential("https", "mygithost")
+                username = LibGit2.filter_username(cfg, cred)
+                @test isnull(username)
+
+                # Add a global credential setting after the URL specific setting. The first
+                # setting to match will be the one that is used.
+                LibGit2.set!(cfg, "credential.username", "bar")
+
+                cred = LibGit2.GitCredential("https", "github.com")
+                username = LibGit2.filter_username(cfg, cred)
+                @test !isnull(username)
+                @test get(username) == "foo"
+
+                cred = LibGit2.GitCredential("https", "mygithost")
+                username = LibGit2.filter_username(cfg, cred)
+                @test !isnull(username)
+                @test get(username) == "bar"
+            finally
+                close(cfg)
+                rm(config_path)
+            end
+        end
+
+        @testset "empty username" begin
+            config_path = joinpath(dir, config_file)
+            cfg = LibGit2.GitConfig(config_path, LibGit2.Consts.CONFIG_LEVEL_APP)
+
+            try
+                # No credential settings should be set for these tests
+                @test isempty(collect(LibGit2.GitConfigIter(cfg, r"credential.*")))
+
+                # An empty username should count as being set
+                LibGit2.set!(cfg, "credential.https://github.com.username", "")
+                LibGit2.set!(cfg, "credential.username", "name")
+
+                cred = LibGit2.GitCredential("https", "github.com")
+                username = LibGit2.filter_username(cfg, cred)
+                @test !isnull(username)
+                @test get(username) == ""
+
+                cred = LibGit2.GitCredential("https", "mygithost", "path")
+                username = LibGit2.filter_username(cfg, cred)
+                @test !isnull(username)
+                @test get(username) == "name"
+            finally
+                close(cfg)
+                rm(config_path)
+            end
+        end
+    end
+
+    @testset "GitCredentialHelpers" begin
+        # In order to use the "store" credential helper `git` needs to be installed and
+        # on the path.
+        git_installed = try
+            success(`git --version`)
+        catch
+            false
+        end
```

Comment:
I could revise these tests not to use `git`. That would involve making a custom helper just for these tests.

Reply:
Hmmm, I don't think that's necessary as I'm guessing `git` is already required for building and testing julia. I'm guessing it should probably be explicitly listed in the [README](https://github.com/JuliaLang/julia#required-build-tools-and-external-libraries) though?

Reply:
`git` isn't a requirement for building Julia. I think you could just the GitHub "Download Zip" link to build Julia without `git`.